portal/gatsby-node.js

```diff
@@ -1,62 +1,253 @@
-// eslint-disable-next-line
+/* eslint-disable @typescript-eslint/no-var-requires */
+const ts = require("typescript");
 const docgen = require("react-docgen-typescript");
+const { createFilePath } = require("gatsby-source-filesystem");
 
-const options = {
+const docgenOptions = {
     savePropValueAsString: true,
     skipChildrenPropWithoutDoc: false,
 };
 
-exports.onCreatePage = async ({ page, actions }) => {
-    const { createPage, deletePage } = actions;
-    const componentPageMatch = page.path.match(/\/.*react.*\/documentation\/(.*)\//);
-    const corePageMatch = page.path.match(/\/core\/documentation\/(.*)\//);
-    const utilPageMatch = page.path.match(/\/.*util.*\/documentation\/(.*)\//);
-    const pageSpecifiedPath = page.context.frontmatter && page.context.frontmatter.path;
-
-    const makePage = (path, withType = false) => {
-        const type = withType
-            ? docgen.parse(`../packages/${page.context.frontmatter.react}/src/index.ts`, options)
-            : undefined;
-
-        const newPage = {
-            ...page,
-            path,
-            context: {
-                ...page.context,
-                frontmatter: {
-                    ...page.context.frontmatter,
-                    type,
-                },
-            },
-        };
-        createPage(newPage);
-        deletePage(page);
-    };
-
-    // If page is a component doc page, create simpler path
-    if (componentPageMatch) {
-        makePage(`/komponenter/${componentPageMatch[1].toLowerCase()}`, true);
-    }
-    if (corePageMatch) {
-        makePage(`/komponenter/${corePageMatch[1].toLowerCase()}`, true);
-    }
-    if (utilPageMatch) {
-        makePage(`/komponenter/${utilPageMatch[1].toLowerCase()}`, true);
-    }
+const tsOptions = {
+    jsx: ts.JsxEmit.ReactJSX,
+    module: ts.ModuleKind.CommonJS,
+    target: ts.ScriptTarget.ES2017,
+    noImplicitAny: true,
+    noImplicitReturns: true,
+    strictNullChecks: true,
+    strictFunctionTypes: true,
+    strictBindCallApply: true,
+    strictPropertyInitialization: true,
+    esModuleInterop: true,
+    noImplicitThis: true,
+    alwaysStrict: true,
+    allowJs: true,
+    checkJs: true,
+};
+
+function parseFiles(fileNames) {
+    const parser = docgen.withDefaultConfig(docgenOptions);
+    const host = ts.createCompilerHost(tsOptions);
+    const program = ts.createProgram(fileNames, tsOptions, host);
+    return parser.parseWithProgramProvider(fileNames, () => program);
```

Comment:
Setter opp én instans av TypeScript som parser alt i én runde i stedet for å starte TS ca 100 ganger for en parsing per fil. Sparer oss for mye tid i bygget.